src/reconcile/rec_write.c

```diff
@@ -2692,43 +2510,39 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 	__rec_dictionary_reset(r);
 
 	/* Set the number of entries and size for the just finished chunk. */
-	last = &r->bnd[r->bnd_next];
-	last->max_bnd_entries = r->entries;
-	last->size = (uint32_t)inuse;
+	r->cur_ptr->entries = r->entries;
+	r->cur_ptr->image.size = inuse;
 
 	/*
 	 * In case of bulk load, write out chunks as we get them. Otherwise we
 	 * keep two chunks in memory at a given time. So, if there is a previous
 	 * chunk, write it out, making space in the buffer for the next chunk to
 	 * be written.
 	 */
-	if (r->is_bulk_load) {
-		dsk->recno = last->max_bnd_recno;
-		dsk->u.entries = last->max_bnd_entries;
-		dsk->mem_size = (uint32_t)inuse;
-		r->cur_img_ptr->size = dsk->mem_size;
-		WT_RET(__rec_split_write(session,
-		    r, last, r->cur_img_ptr, false));
-	} else {
-		WT_RET(__rec_split_write_prev_and_swap_buf(session, r));
-		/* current image we are writing to has changed */
-		dsk = r->cur_img_ptr->mem;
-	}
+	if (r->is_bulk_load)
+		WT_RET(__rec_split_write(session, r, r->cur_ptr, NULL, false));
+	else  {
+		if (r->prev_ptr == NULL) {
+			WT_RET(__rec_split_chunk_init(
+			    session, r, &r->chunkB, r->cur_ptr->image.memsize));
+			r->prev_ptr = &r->chunkB;
+		} else
+			WT_RET(__rec_split_write(
+			    session, r, r->prev_ptr, NULL, false));
 
-	/* Fix where free points */
-	r->first_free = WT_PAGE_HEADER_BYTE(btree, dsk);
+		/* Switch chunks. */
+		tmp = r->prev_ptr;
+		r->prev_ptr = r->cur_ptr;
+		r->cur_ptr = tmp;
+	}
 
-	/* Prepare the next boundary */
-	WT_RET(__rec_split_bnd_grow(session, r));
-	r->bnd_next++;
-	next = &r->bnd[r->bnd_next];
-	/* Set the key for the next chunk. */
-	next->max_bnd_recno = r->recno;
-	if (dsk->type == WT_PAGE_ROW_INT || dsk->type == WT_PAGE_ROW_LEAF)
-		WT_RET(__rec_split_row_promote(
-		    session, r, &next->max_bnd_key, dsk->type));
+	/* Initialize the next chunk. */
+	WT_RET(__rec_split_chunk_init(session, r, r->cur_ptr, 0));
```

Comment:
> This call will reset all chunk members, and will set the chunk's image size to be 0 again. It will not alloc/realloc any memory for the chunk's image.

Correct -- each time we reuse a chunk, we reuse the key and image memory, and chunk-init is initializing the next chunk that we're going to fill in. It's a merge of `__rec_split_bnd_init()` from the old version of the code, and the inline code you added to initialize chunk images.